cumulusci/utils/salesforce/count_sobjects.py

```diff
@@ -0,0 +1,46 @@
+import typing as T
+
+from simple_salesforce import Salesforce
+
+from cumulusci.utils.http.multi_request import CompositeParallelSalesforce
+from cumulusci.utils.iterators import partition
+
+
+class ObjectCount(T.NamedTuple):
+    counts: T.Dict[str, int]
+    transport_errors: T.Sequence[dict]  # e.g. couldn't reach server
+    salesforce_errors: T.Sequence[dict]  # e.g. 404, 401
+
+
+def count_sobjects(sf: Salesforce, objs: T.Sequence[str]) -> ObjectCount:
+    """Quickly count SObjects using SOQL and Parallelization"""
+    with CompositeParallelSalesforce(sf, max_workers=8, chunk_size=5) as cpsf:
+        responses, transport_errors = cpsf.do_composite_requests(
+            (
+                {
+                    "method": "GET",
+                    "url": f"/services/data/v{sf.sf_version}/query/?q=select count() from {obj}",
+                    "referenceId": f"ref{obj}",
```

Comment:
Does Composite actually require that reference ids start with `ref`? Cannot use the raw sObject name?

Reply:
No it's not a rule, just a convention used widely in the docs. Makes it slightly easier to read wire JSON if it is clear when we are referring to an SObject and when to a referenced record. Not opposed to getting rid of it if you would rather save the bytes.